dbt_subprojects/solana/models/_sector/dex/pumpswap/pumpswap_solana_base_trades.sql

```diff
@@ -24,99 +24,206 @@ WITH pools AS (
-        bytearray_to_uint256(bytearray_reverse(bytearray_substring(data, 65, 8))) AS pool_quote_token_reserves,
-        bytearray_to_uint256(bytearray_reverse(bytearray_substring(data, 73, 8))) AS quote_amount_in, -- From Anchor log: quoteAmountIn
-        bytearray_to_uint256(bytearray_reverse(bytearray_substring(data, 89, 8))) AS lp_fee, 
-        bytearray_to_uint256(bytearray_reverse(bytearray_substring(data, 97, 8))) AS protocol_fee_basis_points,
-        bytearray_to_uint256(bytearray_reverse(bytearray_substring(data, 105, 8))) AS protocol_fee,
-        bytearray_to_uint256(bytearray_reverse(bytearray_substring(data, 113, 8))) AS quote_amount_with_lp_fee,
-        bytearray_to_uint256(bytearray_reverse(bytearray_substring(data, 121, 8))) AS user_quote_amount_in, -- From Anchor log: userQuoteAmountIn
-        to_base58(bytearray_substring(data, 129, 32)) AS pool,
-        to_base58(bytearray_substring(data, 161, 32)) AS user,
-        to_base58(bytearray_substring(data, 193, 32)) AS user_base_token_account,
-        to_base58(bytearray_substring(data, 225, 32)) AS user_quote_token_account,
-        to_base58(bytearray_substring(data, 257, 32)) AS protocol_fee_recipient,
-        to_base58(bytearray_substring(data, 289, 32)) AS protocol_fee_recipient_token_account,
-        outer_executing_account
-    FROM {{ source('solana','instruction_calls') }}
-    WHERE inner_executing_account = 'pAMMBay6oceH9fJKBRHGP5D4bD4sWpmSwMn52FMfXEA'
-        AND tx_success = true
-        AND (
-            varbinary_starts_with(data, 0xe445a52e51cb9a1d3e2f370aa503dc2a) OR 
-            varbinary_starts_with(data, 0xe445a52e51cb9a1d67f4521f2cf57777)
-        )
+        token_mint_address,
+        to_token_account,
+        from_token_account,
+        MAX(amount) as amount  -- Just take the largest amount if there are multiple
+    FROM {{ ref('tokens_solana_transfers') }}
+    WHERE token_mint_address = 'So11111111111111111111111111111111111111112'
+    AND token_version = 'spl_token'
     {% if is_incremental() %}
     AND {{incremental_predicate('block_time')}}
+    {% else %}
+    AND block_time >= TIMESTAMP '{{project_start_date}}'
     {% endif %}
+    GROUP BY 
+        tx_id,
+        outer_instruction_index,
+        token_mint_address,
+        to_token_account,
+        from_token_account
+),
+
+swaps_with_transfers AS (
+    SELECT
+        s.*,
+        f.total_fee_rate,
+        CASE
+            WHEN s.is_buy = 1 AND t.to_token_account = s.sol_target_account THEN t.amount
+            WHEN s.is_buy = 0 AND t.from_token_account = s.sol_source_account AND t.to_token_account = s.sol_dest_account THEN t.amount
+            ELSE s.max_min_sol_amount
+        END as sol_amount
+    FROM swaps_base s
+    CROSS JOIN latest_fee_config f
+    LEFT JOIN transfers_aggregated t
+        ON t.tx_id = s.tx_id 
+        AND t.outer_instruction_index = s.outer_instruction_index
```

Comment:
you'll have to add inner instruction to this, maybe like this:

```sql
AND t.inner_instruction_index IN (s.inner_instruction_index +1, s.inner_instruction_index +2 )
```

we have a few other models that work similarly, can study that approach in there.